pandas/core/indexes/interval.py

```diff
@@ -704,7 +704,7 @@ def get_indexer(
 
         self._check_indexing_method(method)
 
-        if self.is_overlapping:
+        if not self._index_as_unique:
```

Comment:
i would leave this as it more reading friendly

Reply:
ok.  i looked at it and took a minute thinking "wait, why is it different in these two places", thought others might have the same confusion